lib/paperclip/attachment_extensions.rb

```diff
@@ -24,6 +24,39 @@ def reprocess_original!
         flush_deletes
       end
     end
+
+    # We overwrite this method to put a circuit breaker around
+    # calls to object storage, to stop hitting APIs that are slow
+    # to respond or don't respond at all and as such minimize the
+    # impact of object storage outages on application throughput
+    def save
+      circuit_break! do
+        flush_deletes unless @options[:keep_old_files]
+
+        process = only_process
+        @queued_for_write.except!(:original) if process.any? && !process.include?(:original)
+
+        flush_writes
+      end
+
+      @dirty = false
+      true
+    end
```

Comment:
Would it make sense to somehow call the upstream method entirely wrapped in `circuit_break!` instead of redefining it?